index.js

```diff
@@ -21,14 +21,17 @@ var toArray = Array.from || (function (slice) {
 
 // ===================================================================
 
-function eventToPromise (emitter, event, opts) {
-  var ignoreErrors = opts && opts.ignoreErrors
+function eventToPromise (emitter, event, _opts) {
+  var opts = _opts || {}
+  var ignoreErrors = opts.ignoreErrors
+  var errorEvent = opts.error || 'error'
 
   return new Bluebird(function (resolve, reject) {
     var addListener =
+      emitter.once ||
```

Comment:
Your are right, my bad.

Removing the listeners is necessary (especially the `error` listener) because its presence changes Node's behaviour.